atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/sweep/queue/SweepableCells.java

```diff
@@ -0,0 +1,251 @@
+        resultIterator.forEachRemaining(entry -> populateWrites(row, entry, writes));
+
+        return new ArrayList<>(writes.values());
+    }
+
+    void deleteDedicatedRows(ShardAndStrategy shardAndStrategy, long partitionFine) {
+        rangeRequestsDedicatedRows(shardAndStrategy, partitionFine).forEach(this::deleteRange);
+    }
+
+    void deleteNonDedicatedRow(ShardAndStrategy shardAndStrategy, long partitionFine) {
+        deleteRange(rangeRequestNonDedicatedRow(shardAndStrategy, partitionFine));
+    }
+
+    private void addReferenceToDedicatedRows(PartitionInfo info, List<WriteInfo> writes, Map<Cell, byte[]> cells) {
+        addCell(info, WriteReference.DUMMY, false, 0, -numberDedicatedRows(writes), cells);
+    }
+
+    private void addWrite(PartitionInfo info, WriteInfo write, boolean dedicate, long index, Map<Cell, byte[]> cells) {
+        addCell(info, write.writeRef(), dedicate, index / MAX_CELLS_DEDICATED, index % MAX_CELLS_DEDICATED, cells);
+    }
+
+    private void addCell(PartitionInfo info, WriteReference writeRef, boolean dedicate, long dedicatedRow,
+            long index, Map<Cell, byte[]> cells) {
+        SweepableCellsTable.SweepableCellsRow row = computeRow(info, dedicate, dedicatedRow);
+        SweepableCellsTable.SweepableCellsColumnValue colVal = createColVal(info.timestamp(), index, writeRef);
+        cells.put(SweepQueueUtils.toCell(row, colVal), colVal.persistValue());
+    }
+
+    private long numberDedicatedRows(List<WriteInfo> writes) {
+        return 1 + (writes.size() - 1) / MAX_CELLS_DEDICATED;
+    }
+
+    private static long tsMod(long timestamp) {
+        return timestamp % SweepQueueUtils.TS_FINE_GRANULARITY;
+    }
+
+    private SweepableCellsTable.SweepableCellsColumnValue createColVal(long ts, long index, WriteReference writeRef) {
+        SweepableCellsTable.SweepableCellsColumn col = SweepableCellsTable.SweepableCellsColumn.of(tsMod(ts), index);
+        return SweepableCellsTable.SweepableCellsColumnValue.of(col, writeRef);
+    }
+
+    private void populateWrites(SweepableCellsTable.SweepableCellsRow row, Map.Entry<Cell, Value> entry,
+            Map<CellReference, WriteInfo> writes) {
+        SweepableCellsTable.SweepableCellsColumn col = computeColumn(entry);
+
+        if (isReferenceToDedicatedRows(col)) {
+            addWritesFromDedicated(row, col, writes);
+        } else {
+            addWriteFromValue(getTimestamp(row, col), entry.getValue(), writes);
+        }
+    }
+
+    private boolean isReferenceToDedicatedRows(SweepableCellsTable.SweepableCellsColumn col) {
+        return col.getWriteIndex() < 0;
+    }
+
+    private void addWritesFromDedicated(SweepableCellsTable.SweepableCellsRow row,
+            SweepableCellsTable.SweepableCellsColumn col, Map<CellReference, WriteInfo> writes) {
+        List<byte[]> dedicatedRows = computeDedicatedRows(row, col);
+        RowColumnRangeIterator iterator = getColumnRangeAll(dedicatedRows);
```

Comment:
`getColumnRangeAll` name could be clearer. Are we getting all columns from the given dedicated rows?

Reply:
getWithColumnRangeAll?

Reply:
`getAllColumns`? At this level, we don't care that we're doing it with `kvs.getRowsColumnRange`, and in any case that is implied by the return type.